### PR TITLE
fix prometheus metrics + show errors with metrics

### PR DESCRIPTION
**Pull request by gburanov**

### CODE REVIEW

File: prom_metrics.go [hunk 1]
```diff
@@ -254,16 +255,23 @@ func (p *PrometheusMetrics) extractSignature(raw string) (name, labels string, e
...
 	for _, l := range split[1:] {
 		if lSplit = strings.Split(l, "="); len(lSplit) != 2 {
 			return "", "", fmt.Errorf(`bad label "%s" in metric "%s"`, l, raw)
 		}
+
 		if !promMetricLabelRe.MatchString(lSplit[0]) {
 			return "", "", fmt.Errorf(`bad label name "%s" in metric "%s"`, l, raw)
 		}
+		if !promMetricLabelRe.MatchString(lSplit[1]) {
+			err = fmt.Errorf(`bad label value "%s" in metric "%s"`, l, raw)
```
Comment (gburanov) [inline prom_metrics.go:268]: lets consider bad label value as non critical value - and just not output it instead.

File: prom_metrics.go [hunk 2]
```diff
@@ -133,10 +134,9 @@ func (p *PrometheusMetrics) Update() (err error) {
 
 	p.registry.Each(func(s string, i interface{}) {
 		var name, labels string
-		var err1 error
-		if name, labels, err1 = p.extractSignature(s); err1 != nil {
-			failures = append(failures, err1.Error())
-			return
+		var err error
+		if name, labels, err = p.extractSignature(s); err != nil {
+			failures = append(failures, err.Error())
```
Comment (gburanov) [inline prom_metrics.go:139]: if there are both labels and errors in extractSignature - we will show both labels and errors in output

File: prom_metrics.go (hunk 1 above)
Comment (gburanov) [reply prom_metrics.go:268]: error will still be in scrape output as before, and it does not brake pattern

File: prom_metrics_test.go [hunk 3]
```diff
@@ -166,12 +167,26 @@ app_with_label_total{service="test",l1="2"} 5
 			"# ERROR bad label name \"label.1=1\" in metric \"app_1,label.1=1 a_3\"\n",
 		},
 	} {
+
+		t.Run("counter with invalid symbols", func(t *testing.T) {
+			r := metrics.NewRegistry()
+			metrics.GetOrRegisterCounter(`act,handler=click,partner=good counter`, r).Inc(2)
+			metrics.GetOrRegisterCounter(`act,handler=click,partner="Shopee" counter`, r).Inc(2)
+			metrics.GetOrRegisterCounter(`act,handler=click,partner=adx1005https://adclick.g.doubleclick.net/aclk?sa=L,time_since_bid=1h request`, r).Inc(3)
```
Comment (cedrics) [inline prom_metrics_test.go:175]: I can remove this line and the tests still pass. What is this used for?

File: prom_metrics_test.go [hunk 4]
```diff
@@ -166,12 +167,26 @@ app_with_label_total{service="test",l1="2"} 5
 			"# ERROR bad label name \"label.1=1\" in metric \"app_1,label.1=1 a_3\"\n",
 		},
 	} {
+
+		t.Run("counter with invalid symbols", func(t *testing.T) {
```
Comment (cedrics) [inline prom_metrics_test.go:171]: Would be great if the description made clear what the expected outcome is:

```
if a label is invalid it still outputs valid labels and an error for the invalid ones
```

File: prom_metrics_test.go (hunk 3 above)
Comment (gburanov) [reply prom_metrics_test.go:175]: I modified test

File: prom_metrics_test.go (hunk 4 above)
Comment (gburanov) [reply prom_metrics_test.go]: 👍

**Pull request merged**